File: testng.test/src/test/java/com/example/apiTest.java
package com.example;

import io.restassured.RestAssured;
import io.restassured.response.Response;

import java.util.List;
import java.util.Map;

import org.testng.Assert;
import org.testng.annotations.Test;
import io.restassured.path.json.JsonPath;

public class apiTest {

    @Test
    public void testGetEndpoint() {
        RestAssured.baseURI = "https://jsonplaceholder.typicode.com";
        
        Response response = RestAssured.given()
                                        .when()
                                        .get("/posts");
        
        Assert.assertEquals(response.getStatusCode(), 200, "Status code is not 200");
        String responseBody = response.getBody().asString();
        JsonPath jsonPath = new JsonPath(responseBody);
        List<?> responseList = jsonPath.getList("$");
        Assert.assertTrue(responseList instanceof List, "Response data is not an array");

        Map<String, Object> expectedPost = Map.of(
            "userId", 8,
            "id", 73,
            "title", "consequuntur deleniti eos quia temporibus ab aliquid at",
            "body", "voluptatem cumque tenetur consequatur expedita ipsum nemo quia explicabo\naut eum minima consequatur\ntempore cumque quae est et\net in consequuntur voluptatem voluptates aut"
        );

        boolean containsExpectedPost = responseList.stream()
            .anyMatch(post -> post instanceof Map && ((Map<?, ?>) post).equals(expectedPost));
        
        Assert.assertTrue(containsExpectedPost, "The response does not contain the expected post");
     
    }
}

File: xunit.test/api.test.cs
using System;
using System.Net;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using Xunit;

namespace MinimalApiTests
{
    public class ApiTests : IAsyncLifetime
    {
        private HttpClient? _httpClient;
        private HttpResponseMessage? _cachedResponse;
        private const string BaseUrl = "https://jsonplaceholder.typicode.com/posts";

        private readonly Post _expectedPost = new Post
        {
            UserId = 8,
            Id = 73,
            Title = "consequuntur deleniti eos quia temporibus ab aliquid at",
            Body = "voluptatem cumque tenetur consequatur expedita ipsum nemo quia explicabo\naut eum minima consequatur\ntempore cumque quae est et\net in consequuntur voluptatem voluptates aut"
        };

        public async Task InitializeAsync()
        {
            _httpClient = new HttpClient();
            _cachedResponse = await _httpClient.GetAsync(BaseUrl);
        }

        [Fact]
        public void API_Response_StatusCode()
        {
            Assert.NotNull(_cachedResponse);
            Assert.Equal(HttpStatusCode.OK, _cachedResponse?.StatusCode);
        }

        [Fact]
        public async Task Get_Posts_ShouldContainExpectedPost()
        {
            Assert.NotNull(_cachedResponse);
            var responseContent = await _cachedResponse.Content.ReadAsStringAsync();
            Assert.False(string.IsNullOrEmpty(responseContent));

            using (JsonDocument doc = JsonDocument.Parse(responseContent))
            {
                Assert.Equal(JsonValueKind.Array, doc.RootElement.ValueKind);

                var posts = doc.RootElement.EnumerateArray()
                    .Select(p => new Post
                    {
                        UserId = p.GetProperty("userId").GetInt32(),
                        Id = p.GetProperty("id").GetInt32(),
                        Title = p.GetProperty("title").GetString() ?? string.Empty,
                        Body = p.GetProperty("body").GetString() ?? string.Empty
                    })
                    .ToList();

                var foundPost = posts.SingleOrDefault(p => p.Id == _expectedPost.Id);

                Assert.NotNull(foundPost);
                Assert.Equal(_expectedPost.UserId, foundPost.UserId);
                Assert.Equal(_expectedPost.Title, foundPost.Title);
                Assert.Equal(_expectedPost.Body, foundPost.Body);
            }
        }

        public async Task DisposeAsync()
        {
            _httpClient?.Dispose();
            await Task.CompletedTask;
        }

        // Define the Post class matching the JSON structure
        private class Post
        {
            public int UserId { get; set; }
            public int Id { get; set; }
            public string Title { get; set; } = string.Empty;
            public string Body { get; set; } = string.Empty;
        }
    }
}
